enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,487 @@
+    part of the installation configuration before manifest generation
+- Support for Placement Groups for Control Plane Machines
+  - Users may replace their control plane at a later date using future automation if they wish to have their control
+    plane within a placement group
+- Support for moving Machines between placement groups
+
+## Proposal
+
+We will introduce new fields within the `Placement` section of the AWS Machine Provider Config which will allow users
+to specify which placement group to create instances within.
+
+If the placement group does not exist, it will be created based upon the group type and, if applicable partition count provided.
+
+We will also allow end users to specify which partition their instances should be created within when appropriate.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I would like to use a spread placement group to ensure that my critical
+  workloads are unlikely to suffer from hardware failures simultaneously.
+- As an operator of High-Performance Computing workloads on OpenShift, I would like to use a cluster placement group to
+  ensure that there is
+  minimal latency between instances within my cluster, allowing maximal throughput of my workloads.
+- As an operator of a large scale distributed and replicated data store, I would like to use a partition placement
+  group to ensure that different shards of my data are replicated within different fault domains within the same
+  availability zone.
+- As an OpenShift cluster administrator, I would like unneeded placement groups to be cleaned up so that I am not
+  polluting my environment with resources that are no longer in use.
+
+### API Extensions
+
+The new fields for this enhancement will be limited to the `Placement` struct within the `AWSMachineProviderConfig` in
+the Machine v1beta1 API group.
+
+The changes are outlined below:
+
+```golang
+type Placement struct {
+  // Existing Fields
+  ...
+
+  // GroupName specifies the name of an AWS placement group to create the Machine within.
+  // If the group specified does not exist, it will be created based on the GroupType
+  // paramenter.
+  // When a Machine is deleted, if this leaves an empty Placement Group, the group is also
+  // deleted.
+  // +optional
+  GroupName string `json:"groupName,omitempty"`
+
+  // GroupType specifies the type of AWS placement group to use for this Machine.
+  // This parameter is only used when a Machine is being created and the named
+  // placement group does not exist.
+  // Valid values are "Cluster", "Partition", "Spread", and omitted, which means
+  // no opinion and the platform chooses a good default which may change over time.
+  // The current default value is "Spread".
+  // +kubebuilder:validation:Enum:="Cluster";"Partition";"Spread"
+  // +optional
+  GroupType AWSPlacementGroupType `json:"groupType,omitempty"`
```

Comment:
since these are embedded, how will you ensure that conflicting names are not chosen.  if conflicting names do appear in the API, how will you resolve it?

Reply:
(For context, this was meant to be on the name)
It is valid for a user to want to share the group name across multiple machinesets, that's a user case for partition placement groups for example.

What we had been considering, though I'm not sure I've noted it, is to, if the placement group type is set, we should check the placement group if it exists, matches the type. So if a user sets the name to my-spread-placement-group and then sets the type to partition, we would error saying that the placement group specified does not match the expected group type.

Do you think that's worthwhile from a UX perspective?